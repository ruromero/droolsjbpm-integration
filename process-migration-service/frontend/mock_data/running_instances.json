[
    {
      "id": 1,
      "processInstanceId": 2,
      "name": "Evaluation",
      "description": "mockup_Evaluation",
      "state": 1,
      "startTime": "2019-01-06T21:06:55:123Z"
    },
    {
      "id": 2,
      "processInstanceId": 3,
      "name": "Evaluation",
      "description": "mockup_Evaluation",
      "state": 1,
      "startTime": "2019-01-06T21:30:23:123Z"
    },
    {
      "id": 3,
      "processInstanceId": 4,
      "name": "Evaluation",
      "description": "mockup_Evaluation",
      "state": 1,
      "startTime": "2019-01-06T21:49:27:123Z"
    },
    {
      "id": 4,
      "processInstanceId": 5,
      "name": "Evaluation",
      "description": "mockup_Evaluation",
      "state": 1,
      "startTime": "2019-01-06T21:49:34:123Z"
    },
    {
      "id": 5,
      "processInstanceId": 6,
      "name": "Evaluation",
      "description": "mockup_Evaluation",
      "state": 1,
      "startTime": "2019-01-06T21:58:54:123Z"
    },
    {
      "id": 6,
      "processInstanceId": 7,
      "name": "Evaluation",
      "description": "mockup_Evaluation",
      "state": 1,
      "startTime": "2019-01-06T21:59:04:123Z"
    }
  ]