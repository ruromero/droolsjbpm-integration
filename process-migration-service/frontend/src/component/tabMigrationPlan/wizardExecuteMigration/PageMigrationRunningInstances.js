import React from "react";

import { orderBy } from "lodash";
import classNames from "classnames";
import * as sort from "sortabular";
import * as resolve from "table-resolver";
import { compose } from "recompose";

import PropTypes from "prop-types";

import {
  customHeaderFormattersDefinition,
  defaultSortingOrder,
  selectionCellFormatter,
  selectionHeaderCellFormatter,
  sortableHeaderCellFormatter,
  tableCellFormatter,
  Table,
  TABLE_SORT_DIRECTION
} from "patternfly-react";
import { Grid } from "patternfly-react";
import { PaginationRow, paginate, PAGINATION_VIEW } from "patternfly-react";

export default class PageMigrationRunningInstances extends React.Component {
  constructor(props) {
    super(props);

    const getSortingColumns = () => this.state.sortingColumns || {};

    const sortableTransform = sort.sort({
      getSortingColumns,
      onSort: selectedColumn => {
        this.setState({
          sortingColumns: sort.byColumn({
            sortingColumns: this.state.sortingColumns,
            sortingOrder: defaultSortingOrder,
            selectedColumn
          })
        });
      },
      // Use property or index dependening on the sortingColumns structure specified
      strategy: sort.strategies.byProperty
    });

    const sortingFormatter = sort.header({
      sortableTransform,
      getSortingColumns,
      strategy: sort.strategies.byProperty
    });

    const stateFormatter = value => {
      const formatState = state => {
        switch (state) {
          case 0:
            return "Pending";
          case 1:
            return "Active";
          case 2:
            return "Completed";
          case 3:
            return "Aborted";
          case 4:
            return "Suspended";
          default:
            return "Other";
        }
      };
      return <Table.Cell>{formatState(value)}</Table.Cell>;
    };

    // enables our custom header formatters extensions to reactabular
    this.customHeaderFormatters = customHeaderFormattersDefinition;
    this.state = {
      // Sort the first column in an ascending way by default.
      sortingColumns: {
        name: {
          direction: TABLE_SORT_DIRECTION.ASC,
          position: 0
        }
      },

      // column definitions
      columns: [
        {
          property: "select",
          header: {
            label: "Select all rows",
            props: {
              index: 0,
              rowSpan: 1,
              colSpan: 1,
              id: "SelectId"
            },
            customFormatters: [selectionHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 0
            },
            formatters: [
              (value, { rowData, rowIndex }) =>
                selectionCellFormatter(
                  { rowData, rowIndex },
                  this.onSelectRow,
                  `${rowIndex}`,
                  `${rowIndex}`
                )
            ]
          }
        },
        {
          property: "processInstanceId",
          header: {
            label: "ID",
            props: {
              index: 1,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [sortableTransform],
            formatters: [sortingFormatter],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 1
            },
            formatters: [tableCellFormatter]
          }
        },
        {
          property: "name",
          header: {
            label: "Name",
            props: {
              index: 2,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [sortableTransform],
            formatters: [sortingFormatter],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 2
            },
            formatters: [tableCellFormatter]
          }
        },
        {
          property: "description",
          header: {
            label: "Description",
            props: {
              index: 3,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [sortableTransform],
            formatters: [sortingFormatter],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 3
            },
            formatters: [tableCellFormatter]
          }
        },
        {
          property: "startTime",
          header: {
            label: "Start Time",
            props: {
              index: 4,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [sortableTransform],
            formatters: [sortingFormatter],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 4
            },
            formatters: [tableCellFormatter]
          }
        },
        {
          property: "state",
          header: {
            label: "State",
            props: {
              index: 5,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [sortableTransform],
            formatters: [sortingFormatter],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 5
            },
            formatters: [stateFormatter]
          }
        }
      ],

      // rows and row selection state
      rows: this.props.runningInstances,
      selectedRows: [],

      // pagination default states
      pagination: {
        page: 1,
        perPage: 6,
        perPageOptions: [6, 10, 15]
      },

      // page input value
      pageChangeValue: 1
    };
  }

  deselectRow = row => {
    return Object.assign({}, row, { selected: false });
  };

  selectRow = row => {
    return Object.assign({}, row, { selected: true });
  };

  onFirstPage = () => {
    this.setPage(1);
  };
  onLastPage = () => {
    const { page } = this.state.pagination;
    const totalPages = this.totalPages();
    if (page < totalPages) {
      this.setPage(totalPages);
    }
  };
  onNextPage = () => {
    const { page } = this.state.pagination;
    if (page < this.totalPages()) {
      this.setPage(this.state.pagination.page + 1);
    }
  };
  onPageInput = e => {
    this.setState({ pageChangeValue: e.target.value });
  };
  onPerPageSelect = eventKey => {
    const newPaginationState = Object.assign({}, this.state.pagination);
    newPaginationState.perPage = eventKey;
    newPaginationState.page = 1;
    this.setState({ pagination: newPaginationState });
  };
  onPreviousPage = () => {
    if (this.state.pagination.page > 1) {
      this.setPage(this.state.pagination.page - 1);
    }
  };
  onRow = row => {
    const { selectedRows } = this.state;
    const selected = selectedRows.indexOf(row.id) > -1;
    return {
      className: classNames({ selected }),
      role: "row"
    };
  };
  onSelectAllRows = event => {
    const { rows, selectedRows } = this.state;
    const { checked } = event.target;
    const currentRows = this.currentRows().rows;

    if (checked) {
      const updatedSelections = [
        ...new Set([...currentRows.map(r => r.id), ...selectedRows])
      ];
      const updatedRows = rows.map(r =>
        updatedSelections.indexOf(r.id) > -1 ? this.selectRow(r) : r
      );
      this.setState({
        // important: you must update rows to force a re-render and trigger onRow hook
        rows: updatedRows,
        selectedRows: updatedSelections
      });
      this.updateSelectedProcessIds(rows, updatedSelections);
    } else {
      const ids = currentRows.map(r => r.id);
      const updatedSelections = selectedRows.filter(
        r => !(ids.indexOf(r) > -1)
      );
      const updatedRows = rows.map(r =>
        updatedSelections.indexOf(r.id) > -1 ? r : this.deselectRow(r)
      );
      this.setState({
        rows: updatedRows,
        selectedRows: updatedSelections
      });

      this.updateSelectedProcessIds(rows, updatedSelections);
    }
  };

  updateSelectedProcessIds = (rows, selectedRows) => {
    this.props.setRunningInstancesIds(selectedRows);
    this.props.onIsValid(selectedRows.length > 0);
  };

  onSelectRow = (event, row) => {
    const { rows, selectedRows } = this.state;
    const selectedRowIndex = rows.findIndex(r => r.id === row.id);
    if (selectedRowIndex > -1) {
      let updatedSelectedRows;
      let updatedRow;
      if (row.selected) {
        updatedSelectedRows = selectedRows.filter(
          r => !(r === row.processInstanceId)
        );
        updatedRow = this.deselectRow(row);
      } else {
        selectedRows.push(row.processInstanceId);
        updatedSelectedRows = selectedRows;
        updatedRow = this.selectRow(row);
      }
      rows[selectedRowIndex] = updatedRow;
      this.setState({
        rows,
        selectedRows: updatedSelectedRows
      });
      this.updateSelectedProcessIds(rows, updatedSelectedRows);
    }
  };

  onSubmit = () => {
    this.setPage(this.state.pageChangeValue);
  };
  setPage = value => {
    const page = Number(value);
    if (
      !Number.isNaN(value) &&
      value !== "" &&
      page > 0 &&
      page <= this.totalPages()
    ) {
      const newPaginationState = Object.assign({}, this.state.pagination);
      newPaginationState.page = page;
      this.setState({ pagination: newPaginationState, pageChangeValue: page });
    }
  };
  currentRows() {
    const { rows, sortingColumns, columns, pagination } = this.state;
    return compose(
      paginate(pagination),
      sort.sorter({
        columns,
        sortingColumns,
        sort: orderBy,
        strategy: sort.strategies.byProperty
      })
    )(rows);
  }
  totalPages = () => {
    const { perPage } = this.state.pagination;
    return Math.ceil(this.props.runningInstances.length / perPage);
  };
  render() {
    const { columns, pagination, sortingColumns, pageChangeValue } = this.state;
    const sortedPaginatedRows = this.currentRows();

    return (
      <Grid fluid>
        <Table.PfProvider
          striped
          bordered
          hover
          dataTable
          columns={columns}
          components={{
            header: {
              cell: cellProps =>
                this.customHeaderFormatters({
                  cellProps,
                  columns,
                  sortingColumns,
                  rows: sortedPaginatedRows.rows,
                  onSelectAllRows: this.onSelectAllRows
                })
            }
          }}
        >
          <Table.Header headerRows={resolve.headerRows({ columns })} />
          <Table.Body
            rows={sortedPaginatedRows.rows}
            rowKey="processInstanceId"
            onRow={this.onRow}
          />
        </Table.PfProvider>
        <PaginationRow
          viewType={PAGINATION_VIEW.TABLE}
          pagination={pagination}
          pageInputValue={pageChangeValue}
          amountOfPages={sortedPaginatedRows.amountOfPages}
          itemCount={sortedPaginatedRows.itemCount}
          itemsStart={sortedPaginatedRows.itemsStart}
          itemsEnd={sortedPaginatedRows.itemsEnd}
          onPerPageSelect={this.onPerPageSelect}
          onFirstPage={this.onFirstPage}
          onPreviousPage={this.onPreviousPage}
          onPageInput={this.onPageInput}
          onNextPage={this.onNextPage}
          onLastPage={this.onLastPage}
          onSubmit={this.onSubmit}
        />
      </Grid>
    );
  }
}

PageMigrationRunningInstances.propTypes = {
  runningInstances: PropTypes.array.isRequired,
  setRunningInstancesIds: PropTypes.func.isRequired,
  onIsValid: PropTypes.func.isRequired
};
