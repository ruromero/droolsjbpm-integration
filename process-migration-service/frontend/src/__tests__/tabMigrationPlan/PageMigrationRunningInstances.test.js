import renderer from "react-test-renderer";
import React from "react";
import PageMigrationRunningInstances from "../../component/tabMigrationPlan/wizardExecuteMigration/PageMigrationRunningInstances";
import running_instances from "../../../mock_data/running_instances.json";

test("PageMigrationRunningInstances renders correctly using snapshot", () => {
  const myMock = jest.fn();
  const tree = renderer
    .create(
      <PageMigrationRunningInstances
        runningInstances={running_instances}
        setRunningInstancesIds={myMock}
        onIsValid={myMock}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
