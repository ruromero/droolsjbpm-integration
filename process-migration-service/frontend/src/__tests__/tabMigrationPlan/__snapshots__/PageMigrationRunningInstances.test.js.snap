// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMigrationRunningInstances renders correctly using snapshot 1`] = `
<div
  className="container-fluid"
>
  <table
    className="table dataTable table-striped table-bordered table-hover"
    role="grid"
  >
    <thead>
      <tr>
        <th
          aria-label="Select all rows"
          className="table-view-pf-select"
          colSpan={1}
          id="SelectId"
          index={0}
          rowSpan={1}
        >
          <label
            className="control-label sr-only"
            htmlFor="SelectId"
          >
            Select all rows
          </label>
          <input
            checked={false}
            id="SelectId"
            onChange={[Function]}
            type="checkbox"
          />
        </th>
        <th
          aria-label="ID"
          className="sorting sort sort-none"
          colSpan={1}
          index={1}
          onClick={[Function]}
          rowSpan={1}
        >
          ID
        </th>
        <th
          aria-label="Name"
          className="sorting_asc sort sort-asc"
          colSpan={1}
          index={2}
          onClick={[Function]}
          rowSpan={1}
        >
          Name
        </th>
        <th
          aria-label="Description"
          className="sorting sort sort-none"
          colSpan={1}
          index={3}
          onClick={[Function]}
          rowSpan={1}
        >
          Description
        </th>
        <th
          aria-label="Start Time"
          className="sorting sort sort-none"
          colSpan={1}
          index={4}
          onClick={[Function]}
          rowSpan={1}
        >
          Start Time
        </th>
        <th
          aria-label="State"
          className="sorting sort sort-none"
          colSpan={1}
          index={5}
          onClick={[Function]}
          rowSpan={1}
        >
          State
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className=""
        role="row"
      >
        <td
          className="table-view-pf-select"
        >
          <label
            className="control-label sr-only"
            htmlFor="0"
          >
            0
          </label>
          <input
            checked={false}
            id="0"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className=""
        >
          2
        </td>
        <td
          className=""
        >
          Evaluation
        </td>
        <td
          className=""
        >
          mockup_Evaluation
        </td>
        <td
          className=""
        >
          2019-01-06T21:06:55:123Z
        </td>
        <td
          className=""
        >
          Active
        </td>
      </tr>
      <tr
        className=""
        role="row"
      >
        <td
          className="table-view-pf-select"
        >
          <label
            className="control-label sr-only"
            htmlFor="1"
          >
            1
          </label>
          <input
            checked={false}
            id="1"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className=""
        >
          3
        </td>
        <td
          className=""
        >
          Evaluation
        </td>
        <td
          className=""
        >
          mockup_Evaluation
        </td>
        <td
          className=""
        >
          2019-01-06T21:30:23:123Z
        </td>
        <td
          className=""
        >
          Active
        </td>
      </tr>
      <tr
        className=""
        role="row"
      >
        <td
          className="table-view-pf-select"
        >
          <label
            className="control-label sr-only"
            htmlFor="2"
          >
            2
          </label>
          <input
            checked={false}
            id="2"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className=""
        >
          4
        </td>
        <td
          className=""
        >
          Evaluation
        </td>
        <td
          className=""
        >
          mockup_Evaluation
        </td>
        <td
          className=""
        >
          2019-01-06T21:49:27:123Z
        </td>
        <td
          className=""
        >
          Active
        </td>
      </tr>
      <tr
        className=""
        role="row"
      >
        <td
          className="table-view-pf-select"
        >
          <label
            className="control-label sr-only"
            htmlFor="3"
          >
            3
          </label>
          <input
            checked={false}
            id="3"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className=""
        >
          5
        </td>
        <td
          className=""
        >
          Evaluation
        </td>
        <td
          className=""
        >
          mockup_Evaluation
        </td>
        <td
          className=""
        >
          2019-01-06T21:49:34:123Z
        </td>
        <td
          className=""
        >
          Active
        </td>
      </tr>
      <tr
        className=""
        role="row"
      >
        <td
          className="table-view-pf-select"
        >
          <label
            className="control-label sr-only"
            htmlFor="4"
          >
            4
          </label>
          <input
            checked={false}
            id="4"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className=""
        >
          6
        </td>
        <td
          className=""
        >
          Evaluation
        </td>
        <td
          className=""
        >
          mockup_Evaluation
        </td>
        <td
          className=""
        >
          2019-01-06T21:58:54:123Z
        </td>
        <td
          className=""
        >
          Active
        </td>
      </tr>
      <tr
        className=""
        role="row"
      >
        <td
          className="table-view-pf-select"
        >
          <label
            className="control-label sr-only"
            htmlFor="5"
          >
            5
          </label>
          <input
            checked={false}
            id="5"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className=""
        >
          7
        </td>
        <td
          className=""
        >
          Evaluation
        </td>
        <td
          className=""
        >
          mockup_Evaluation
        </td>
        <td
          className=""
        >
          2019-01-06T21:59:04:123Z
        </td>
        <td
          className=""
        >
          Active
        </td>
      </tr>
    </tbody>
  </table>
  <form
    className="content-view-pf-pagination table-view-pf-pagination clearfix"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <div
        className="dropup pagination-pf-pagesize btn-group"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-default"
          disabled={false}
          id="pagination-row-dropdown"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          type="button"
        >
          6
           
          <span
            className="caret"
          />
        </button>
        <ul
          aria-labelledby="pagination-row-dropdown"
          className="dropdown-menu"
          role="menu"
        >
          <li
            className="active"
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              6
            </a>
          </li>
          <li
            className=""
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              10
            </a>
          </li>
          <li
            className=""
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              15
            </a>
          </li>
        </ul>
      </div>
      <span>
        per page
      </span>
    </div>
    <div
      className="form-group"
    >
      <span>
        <span
          className="pagination-pf-items-current"
        >
          1
          -
          6
        </span>
         
        of
         
        <span
          className="pagination-pf-items-total"
        >
          6
        </span>
      </span>
      <ul
        className="pagination pagination-pf-back"
      >
        <li
          className="disabled"
        >
          <a
            href="#"
            onClick={[Function]}
            title="First Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-double-left i"
            />
          </a>
        </li>
        <li
          className="disabled"
        >
          <a
            href="#"
            onClick={[Function]}
            title="Previous Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-left i"
            />
          </a>
        </li>
      </ul>
      <label
        className="sr-only control-label"
      >
        Current Page
      </label>
      <input
        className="pagination-pf-page form-control"
        onChange={[Function]}
        type="text"
        value={1}
      />
      <span>
         
        of
         
        <span
          className="pagination-pf-pages"
        >
          1
        </span>
      </span>
      <ul
        className="pagination pagination-pf-forward"
      >
        <li
          className="disabled"
        >
          <a
            href="#"
            onClick={[Function]}
            title="Next Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-right i"
            />
          </a>
        </li>
        <li
          className="disabled"
        >
          <a
            href="#"
            onClick={[Function]}
            title="Last Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-double-right i"
            />
          </a>
        </li>
      </ul>
    </div>
  </form>
</div>
`;
